lib/modules/tests/index.js

```diff
@@ -0,0 +1,239 @@
+          });
+        });
+      }
+      cb(null, [filePath]);
+    });
+  }
+
+  runJSTests(files, options, cb) {
+    const self = this;
+    const loglevel = options.loglevel || 'warn';
+    async.waterfall([
+      function setupGlobalNamespace(next) {
+        const test = new Test({loglevel, node: options.node, events: self.events, logger: self.logger,
+          config: self.embark.config, ipc: self.ipc});
+        global.embark = test;
+        global.assert = assert;
+        global.config = test.config.bind(test);
+
+        let deprecatedWarning = function () {
+          self.logger.error(__('%s are not supported anymore', 'EmbarkSpec & deployAll').red);
+          self.logger.error(__('You can learn about the new revamped tests here: %s', 'https://embark.status.im/docs/testing.html'.underline));
+          process.exit();
+        };
+
+        global.deployAll = deprecatedWarning;
+        global.EmbarkSpec = {};
+        global.EmbarkSpec.deployAll = deprecatedWarning;
+
+        // Override require to enable `require('Embark/contracts/contractName');`
+        const Module = require('module');
+        const originalRequire = require('module').prototype.require;
+        Module.prototype.require = function (requireName) {
+          if (requireName.startsWith('Embark/contracts')) {
+            return test.require(...arguments);
+          }
+          return originalRequire.apply(this, arguments);
+        };
+
+        global.contract = function (describeName, callback) {
+          return Mocha.describe(describeName, callback);
+        };
+
+        test.init((err) => {
+          next(err, files);
+        });
+      },
+      function executeForAllFiles(files, next) {
+        let fns = files.map((file) => {
+          return (cb) => {
+            const mocha = new Mocha();
+            mocha.reporter(EmbarkSpec, {
+              events: self.events,
+              gasDetails: options.gasDetails,
+              gasLimit: 6000000
+            });
+
+            mocha.addFile(file);
+            mocha.suite.timeout(0);
+            mocha.suite.beforeAll('Wait for deploy', (done) => {
+              if (global.embark.needConfig) {
+                global.config({});
+              }
+              global.embark.onReady((err) => {
+                done(err);
+              });
+            });
+            mocha.run(function (fails) {
+              mocha.suite.removeAllListeners();
+              // Mocha prints the error already
+              cb(null, fails);
+            });
+          };
+        });
+        async.series(fns, next);
+      }
+    ], (err, runs) => {
+      if(err) {
+        return cb(err);
+      }
+      let failures = runs.reduce((acc, val) => acc + val, 0);
+      fs.remove('.embark/contracts', (_err) => {
+        cb(null, {failures});
+      });
+    });
+  }
+
+  runSolidityTests(files, options, cb) {
+    console.info('Running solc tests');
+    const loglevel = options.loglevel || 'warn';
```

Comment:
Log levels should as well be constants (ideally an enum) exposed by Embark core or something.

Can we put `warn` in a constant for now?